lib/modules/datasource/hexpm-bob/index.ts

```diff
@@ -0,0 +1,180 @@
+            releaseTimestamp: new Date(buildDate).getTime(),
+            version: this.cleanVersion(version, packageType),
+            newDigest,
+            constraints: this.constraints(version, packageType),
+            sourceUrl: this.sourceUrl(gitRef, packageType),
+          };
+        });
+    } catch (err) {
+      if (err instanceof HttpError && err.response?.statusCode !== 404) {
+        throw new ExternalHostError(err);
+      }
+      this.handleGenericErrors(err);
+    }
+
+    return result.releases.length > 0 ? result : null;
+  }
+
+  private getPackageType(packageName: string): PackageType | null {
+    if (packageName === 'elixir') {
+      return 'elixir';
+    }
+    if (packageName.startsWith('otp/')) {
+      return 'erlang';
+    }
+    return null;
+  }
+
+  // eslint-disable-next-line consistent-return
+  private cleanVersion(version: string, packageType: PackageType): string {
+    switch (packageType) {
+      case 'elixir':
+        return version.replace(/^v/, '');
+      case 'erlang':
+        return version.replace(/^OTP-/, '');
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  private sourceUrl(gitRef: string, packageType: PackageType): string {
+    switch (packageType) {
+      case 'elixir':
+        return `https://github.com/elixir-lang/elixir/tree/${gitRef}`;
+      case 'erlang':
+        return `https://github.com/erlang/otp/tree/${gitRef}`;
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  private isStable(version: string, packageType: PackageType): boolean {
+    switch (packageType) {
+      case 'elixir':
+        return version.match(/^v\d+\.\d+\.\d+($|-otp)/) !== null;
+      case 'erlang':
+        return version.startsWith('OTP-');
+    }
+  }
+
+  // eslint-disable-next-line consistent-return
+  private changelogUrl(
+    version: string,
+    packageType: PackageType
+  ): string | undefined {
+    if (this.isStable(version, packageType)) {
+      return undefined;
+    }
+
+    switch (packageType) {
+      case 'elixir':
+        return `https://github.com/elixir-lang/elixir/releases/tag/${version}`;
+      case 'erlang':
+        return `https://github.com/erlang/otp/releases/tag/${version}`;
+    }
+  }
+
+  private constraints(
+    version: string,
+    packageType: PackageType
+  ): Record<string, string[]> | undefined {
+    if (packageType !== 'elixir') {
+      return undefined;
+    }
+
+    const otpRequirement = version.match(/-otp-(?<otpVersion>\d+)$/);
+
+    if (!otpRequirement) {
+      return undefined;
+    }
+
+    return { erlang: [`^${otpRequirement.groups!.otpVersion}.0`] };
+  }
+
+  private getPackageDetails(
+    packageType: PackageType
+  ): Omit<ReleaseResult, 'releases'> {
+    let specificDetails: Partial<ReleaseResult>;
+    switch (packageType) {
+      case 'elixir':
+        specificDetails = {
+          changelogUrl: 'https://github.com/elixir-lang/elixir/releases',
```

Comment:
can be removed, as it should point to the repo. renovate uses sourceUrl if no changelogUrl so can be removed